generator/views/datagroups.py

```diff
@@ -0,0 +1,162 @@
+            dataset_id=table.dataset_id,
+            table_id=table.table_id,
+        ),
+        # Note: Ideally we'd use the table's ETL schedule as the `maximum_cache_age` but we don't have schedule
+        # metadata here.
+    )
+
+
+def _get_datagroup_from_bigquery_view(
+    view: bigquery.Table,
+    client: bigquery.Client,
+    dataset_view_map: Dict[str, BQViewReferenceMap],
+) -> Optional[Datagroup]:
+    # Dataset view map only contains references for shared-prod views.
+    if view.project not in ("moz-fx-data-shared-prod", "mozdata"):
+        logging.debug(
+            f"Unable to get sources for non shared-prod/mozdata view: {view.full_table_id} in generated-sql."
+        )
+        return None
+
+    dataset_view_references = dataset_view_map.get(view.dataset_id)
+    if dataset_view_references is None:
+        logging.debug(f"Unable to find dataset {view.dataset_id} in generated-sql.")
+        return None
+
+    view_references = dataset_view_references.get(view.table_id)
+    if not view_references or len(view_references) > 1:
+        # For views with multiple sources it's unclear which table to check for updates.
+        logging.debug(
+            f"Unable to find a single source for {view.full_table_id} in generated-sql."
+        )
+        return None
+
+    source_table_id = ".".join(view_references[0])
+    try:
+        table = client.get_table(source_table_id)
+        if "TABLE" == table.table_type:
+            return _get_datagroup_from_bigquery_table(table)
+        elif "VIEW" == table.table_type:
+            return _get_datagroup_from_bigquery_view(table, client, dataset_view_map)
+    except NotFound as e:
+        raise ValueError(
+            f"Unable to find {source_table_id} referenced in {view.full_table_id}"
+        ) from e
+
+    return None
+
+
+def _generate_view_datagroup_lkml(
+    view: View,
+    client: bigquery.Client,
+    dataset_view_map: Dict[str, BQViewReferenceMap],
+) -> str:
+    """Generate the Datagroup LookML for a Looker View."""
+    # Only generate datagroup for views that can be linked to a BigQuery table:
+    if view.view_type != TableView.type:
+        return ""
+
+    # Use the release channel table or the first available table (usually the only one):
+    view_table = next(
+        (table for table in view.tables if table.get("channel") == "release"),
+        view.tables[0],
+    )["table"]
+
+    try:
+        bq_table = client.get_table(view_table)
+    except NotFound as e:
+        raise ValueError(
+            f"{view_table} not found when generating datagroup but in namespaces yaml."
+        ) from e
+
+    if "TABLE" == bq_table.table_type:
+        datagroup: Optional[Datagroup] = _get_datagroup_from_bigquery_table(bq_table)
+        return str(datagroup)
+    elif "VIEW" == bq_table.table_type:
+        datagroup = _get_datagroup_from_bigquery_view(
+            bq_table, client, dataset_view_map
+        )
+        if datagroup is not None:
```

Comment:
```suggestion
        if not datagroup:
```

Reply:
as i understand it, `is not None` is actually the python idiomatic way to handle this here.

Reply:
Yup, according to the Google style guide anyway: https://google.github.io/styleguide/pyguide.html#2144-decision (when explicitly checking for `None` which is the intention of `_get_datagroup_from_bigquery_view`

Reply:
my bad, thanks for clarifying.